skglm/solvers/prox_newton.py

```diff
@@ -63,20 +67,37 @@ def solve(self, X, y, datafit, penalty, w_init=None, Xw_init=None):
         if is_sparse:
             X_bundles = (X.data, X.indptr, X.indices)
 
+        if len(w) != n_features + self.fit_intercept:
+            if self.fit_intercept:
+                val_error_message = (
+                    "w should be of size n_features + 1 when using fit_intercept=True: "
+                    f"expected {n_features + 1}, got {len(w)}.")
+            else:
+                val_error_message = (
+                    "w should be of size n_features: "
+                    f"expected {n_features}, got {len(w)}.")
+            raise ValueError(val_error_message)
+
         for t in range(self.max_iter):
             # compute scores
             if is_sparse:
                 grad = _construct_grad_sparse(
-                    *X_bundles, y, w, Xw, datafit, all_features)
+                    *X_bundles, y, w[:n_features], Xw, datafit, all_features)
             else:
-                grad = _construct_grad(X, y, w, Xw, datafit, all_features)
+                grad = _construct_grad(X, y, w[:n_features], Xw, datafit, all_features)
 
-            opt = penalty.subdiff_distance(w, grad, all_features)
+            opt = penalty.subdiff_distance(w[:n_features], grad, all_features)
+
+            # optimality of intercept
+            if fit_intercept:
+                intercept_opt = np.abs(np.sum(datafit.raw_grad(y, Xw)))
```

Comment:
To be sure I understand correctly : summing is like taking dot product with np.ones(n_features), so that's the magnitude of the gradient with respect to a constant feature here ?

Can you add a comment ("gradient wrt intercept" or "gradient with respect to constant feature of 1s") ?